modules/c++/re/include/re/Regex.h

```diff
@@ -0,0 +1,203 @@
+         *  \throw  RegexException on fatal error
+         */
+        bool match(const std::string& str,
+                   RegexMatch& matchObject);
+
+        bool matches(const std::string& str) const;
+
+        /*!
+         *  Search the matchString
+         *  See the pcre man pages (pcre_exec) for further
+         *  info on the optional flags.
+         *  \param matchString The string to try and match
+         *  \param startIndex Starting where?
+         *  \param flags  Any flags
+         *  \return  Matched substring
+         *  \throw  RegexException on fatal error
+         */
+        std::string search(const std::string& matchString,
+                           int startIndex = 0);
+
+        /*!
+         *  Search the matchString and get the sub-expressions, by ref
+         *  \param matchString The string to match
+         *  \param v The sub-expression
+         */
+        void searchAll(const std::string& matchString,
+                       RegexMatch& v);
+
+        /*!
+         *  Split the string by occurrences of the pattern
+         *  \param str  The string to split
+         *  \param v    The resulting container of matches split from str
+         */
+        void split(const std::string& str,
+                   std::vector<std::string>& v);
+
+        /*!
+         *  Replace occurrences of the pattern in the string
+         *  \param str  The string in which to replace the pattern
+         *  \param repl  The replacement
+         *  \return  The resulting string
+         */
+        std::string sub(const std::string& str,
+                        const std::string& repl);
+
+        /*!
+         *  Backslash all non-alphanumeric characters
+         *  \param str  The string to escape
+         *  \return  The escaped string
+         */
+        std::string escape(const std::string& str) const;
+
+    private:
+        std::string mPattern;
+
+#ifdef __CODA_CPP11
+        /*!
+         *  Replace non-escaped "." with "[\s\S]" to get Regex_DOTALL newline behavior
+         *  \param str  The string to modify
+         *  \return  The modified string
+         */
+        std::string replaceDot(const std::string& str) const;
+
+        bool searchWithContext(std::string::const_iterator inputIterBegin,
+                               std::string::const_iterator inputIterEnd,
+                               std::smatch& match,
+                               bool matchBeginning=true) const;
+
+        std::regex mRegex;
+#else
+        // Internal function for passing flags to pcre_exec()
+        std::string search(const std::string& matchString,
+                           int startIndex, int flag);
+
+        // Size of the output vector, must be a multiple of 3
+        // The output vector is filled up to 2/3 (666) full for matches
+        // so the maximum number of substrings is 333 (333 start
+        // offsets and 333 end offsets)
+        const int OVECCOUNT;
```

Comment:
If this is a member variable now (didn't it used to be a static constant?), then it should change over to a normal member variable name like `mOvectorCount`.  But since `mOvector` is an STL vector, can't you just use `mOvector.size()` throughout and get rid of `OVECCOUNT` altogether?